File: Arithmetic/arithmetic.go
package main

import (
	"strconv"
)

func AdditionBigNumbers(a string, b string) string {
	for len(b) < len(a) {
		b = "0" + b
	}
	for len(a) < len(b) {
		a = "0" + a
	}
	result := ""
	c := 0
	for i := len(a) - 1; i >= 0; i-- {
		localResult := int(a[i]-48) + int(b[i]-48) + c
		c = 0
		if localResult > 9 {
			localResult -= 10
			c = 1
		}
		result += strconv.Itoa(localResult)
	}
	if c == 1 {
		result += "1"
	}
	fresult := ""
	for _, s := range result {
		fresult = string(s) + fresult
	}
	return fresult
}

func SubtractionBigNumbers(a string, b string) string {
	k := false
	if len(a) < len(b) {
		a, b = b, a
		k = true
	}
	ai, _ := strconv.Atoi(a)
	bi, _ := strconv.Atoi(b)
	if ai < bi {
		a, b = b, a
		k = true
	}
	for len(b) < len(a) {
		b = "0" + b
	}
	for len(a) < len(b) {
		a = "0" + a
	}
	result := ""
	c := 0
	for i := len(a) - 1; i >= 0; i-- {
		localResult := int(a[i]-48) - int(b[i]-48) - c
		c = 0
		if localResult < 0 {
			localResult += 10
			c = 1
		}
		result += strconv.Itoa(localResult)
	}
	if c == 1 {
		result += "1"
	}
	for result[len(result)-1] == '0' {
		result = result[:len(result)-1]
	}
	fresult := ""
	for _, s := range result {
		fresult = string(s) + fresult
	}
	if k {
		fresult = "-" + fresult
	}
	return fresult
}

func MultiplicationBigNumbers(a string, b string) string {
	result := ""
	c := 0
	for i := len(b) - 1; i >= 0; i-- {
		localResult := ""
		c = 0
		for j := len(a) - 1; j >= 0; j-- {
			localLocalResult := int(a[j]-48)*int(b[i]-48) + c
			c = 0
			for localLocalResult > 9 {
				localLocalResult -= 10
				c++
			}
			localResult += strconv.Itoa(localLocalResult)
		}
		if c > 0 {
			localResult += strconv.Itoa(c)
			c = 0
		}
		flocalResult := ""
		for _, s := range localResult {
			flocalResult = string(s) + flocalResult
		}
		for f := 0; f < len(b)-i-1; f++ {
			flocalResult += "0"
		}
		result = AdditionBigNumbers(result, flocalResult)
	}
	return result
}

func main() {

}

File: Arithmetic/arithmetic.py
def addition_big_numbers(a, b):
    a = str(a)[::-1]
    b = str(b)[::-1]

    while(len(b) < len(a)):
        b += '0'
    while(len(a) < len(b)):
        a += '0'

    result = ''
    c = 0
    for i in range(0, len(a)):
        local_result = int(a[i]) + int(b[i]) + c
        c = 0
        if local_result > 9:
            local_result -= 10
            c = 1
        result += str(local_result)
    if c == 1:
        result += '1'
    return result[::-1]

def subtraction_big_numbers(a, b):
    k = False
    if int(a) > int(b):
        a = str(a)[::-1]
        b = str(b)[::-1]
    else:
        k = True
        swap = str(a)[::-1]
        a = str(b)[::-1]
        b = swap
    while(len(b) < len(a)):
        b += '0'
    while(len(a) < len(b)):
        a += '0'

    result = ''
    c = 0
    for i in range(0, len(a)):
        local_result = int(a[i]) - int(b[i]) - c
        c = 0
        if local_result < 0:
            local_result += 10
            c = 1
        result += str(local_result)

    if k:
        result += '-'
    while result[-1] == '0':
        result = result[:-1]
    return result[::-1]

def multiplication_big_numbers(a, b):
    a = str(a)[::-1]
    b = str(b)[::-1]

    result = ''
    c = 0
    for i in range(0, len(b)):
        local_result = ''
        c = 0
        for j in range(0, len(a)):
            local_local_result = int(a[j]) * int(b[i]) + c
            c = 0
            while local_local_result > 9:
                local_local_result -= 10
                c += 1
            local_result += str(local_local_result)
        if c > 0:
            local_result += str(c)
            c = 0
        result = addition_big_numbers(str(result), str(local_result)[::-1] + ('0' * i))
    return result